platforms/ide/problems/src/main/java/org/gradle/problems/internal/emitters/BuildOperationBasedProblemEmitter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.problems.internal.emitters;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.problems.Problem;
+import org.gradle.api.problems.ProblemEmitter;
+import org.gradle.api.problems.internal.DefaultProblem;
+import org.gradle.api.problems.internal.DefaultProblemProgressDetails;
+import org.gradle.internal.operations.BuildOperationProgressEventEmitter;
+
+/**
+ * Emits problems as build operation progress events.
+ *
+ * @since 8.6
+ */
+@Incubating
+public class BuildOperationBasedProblemEmitter implements ProblemEmitter {
+
+    private final BuildOperationProgressEventEmitter eventEmitter;
+
+    public BuildOperationBasedProblemEmitter(BuildOperationProgressEventEmitter eventEmitter) {
+        this.eventEmitter = eventEmitter;
+    }
+
+    @Override
+    public void emit(Problem problem) {
+        if (problem instanceof DefaultProblem) {
+            DefaultProblem defaultProblem = (DefaultProblem) problem;
+
+            // TODO: Decide to emit a different problem, log, or throw
```

Comment:
@donat I'm to opinions about this

Reply:
My thinking here is that problems should be only emitted here for the time being. We can decide if we want to do something different with reported problems (suppress, log, throw) in problem transformers. There, we can check for the problem category and implement new behaviour.

Reply:
I don't think we talk about the same here: if you don't have a build operation for a problem, something weird is going on. Should we do something when we detect a case like this?

Reply:
The least we should do is to log an error. My gut feeling is that we should not throw an exception here.